x/validators/errors.go

```diff
@@ -1,44 +1,15 @@
 package validators
 
 import (
-	stderr "errors"
-	"reflect"
-
 	"github.com/iov-one/weave/errors"
 )
 
-// ABCI Response Codes
-// x/update_validators reserves 40 ~ 49.
-const (
-	CodeEmptyDiff         = 40
-	CodeWrongType         = 41
-	CodeInvalidPubKey     = 42
-	CodeEmptyValidatorSet = 43
-	CodeInvalidPower      = 44
-	CodeNotFound          = 45
-)
+// Error codes
+// x/update_validators reserves 140 ~ 149.
 
 var (
-	errEmptyDiff         = stderr.New("Empty validator diff")
-	errWrongType         = stderr.New("Wrong type for accounts storage")
-	errInvalidPubKey     = stderr.New("Invalid public key")
-	errEmptyValidatorSet = stderr.New("Empty validator set")
-	errInvalidPower      = stderr.New("Power value is invalid")
-	errNotFound          = stderr.New("Not found")
-)
-
-func ErrEmptyDiff() error {
-	return errors.WithCode(errEmptyDiff, CodeEmptyDiff)
-}
-
-func ErrWrongType(t interface{}) error {
-	typeName := ""
-	if t != nil {
-		typeName = reflect.TypeOf(t).Name()
-	}
-	return errors.WithLog(typeName, errWrongType, CodeWrongType)
-}
-
-func ErrNotFound(entityName string) error {
-	return errors.WithLog(entityName, errNotFound, CodeNotFound)
-}
+	EmptyDiffErr = errors.Register(140, "empty validator diff")
+	InvalidPubKeyErr = errors.Register(141, "invalid public key")
+	EmptyValidatorErr = errors.Register(142, "empty validator set")
```

Comment:
Isn't there a generic `EmptyError`  we can use and wrap, rather than define a new one here? 

I mean, maybe even for diff as well, just:
`errors.EmptyError.New("validator diff")`
`errors.EmptyError.New("validator set")`

Don't we want to remove the overhead of unnecessary package-level errors?

Reply:
We do, I was not really sure about this one